File: Lab5-TabBar&Container/Lab5-TabBar&Container/DemoFile/SecViewController/SecViewController.m
//
//  SecViewController.m
//  Lab5-TabBar&Container
//
//  Created by 張力元 on 2019/2/15.
//  Copyright © 2019 張力元. All rights reserved.
//

#import "SecViewController.h"

@interface SecViewController ()
@property (strong, nonatomic) IBOutlet UIScrollView *scrollView;

@end

@implementation SecViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    NSLog(@"%@ viewDidLoad",[self class]);
}

- (void)viewWillAppear:(BOOL)animated{
    [super viewWillAppear:animated];
    NSLog(@"%@ viewWillAppear",[self class]);
}

- (void)viewDidAppear:(BOOL)animated{
    [super viewDidAppear:animated];
    NSLog(@"%@ viewDidAppear",[self class]);
}

- (void)viewWillDisappear:(BOOL)animated{
    [super viewWillDisappear:animated];
    NSLog(@"%@ viewWillDisappear",[self class]);
}

- (void)viewDidDisappear:(BOOL)animated{
    [super viewDidDisappear:animated];
    NSLog(@"%@ viewDidDisappear",[self class]);
}

@end

File: Swift/lab4/lab4-1-7/delegate/lab4-1-6/SecViewController.swift
//
//  SecViewController.swift
//  lab4-1-6
//
//  Created by taizhou on 2019/8/1.
//  Copyright © 2019 taizhou. All rights reserved.
//

import UIKit

//宣告合約內容
@objc protocol SecViewControllerDelegate {
    //必定要執行的副程式（義務）
    func changeTitle(title:String)
    //選擇性要執行的副程式（權利）
    @objc optional func selectionFunction()
}



class SecViewController: UIViewController {
    weak var delegate : SecViewControllerDelegate?
    
    var data = ""
    
    @IBOutlet var showDataLabel: UILabel!
    @IBOutlet var textField: UITextField!
    
    
    
    
    override func viewDidLoad() {
        super.viewDidLoad()

        showDataLabel.text = data
    }
    
    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(animated)
        //宣告一個欲回傳的字串為lastTitle
        let lastTitle = "新標題"
        //透過合約內的「changeTitle:」副程式將lastTitle傳回ViewController頁面
        delegate?.changeTitle(title: lastTitle)
    }
    
}

